packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -590,7 +590,7 @@ class HotRunner extends ResidentRunner {
           pause: pause
         );
         countExpectedReports += reports.length;
-        Future.wait(reports).catchError((dynamic error) {
+        await Future.wait(reports).catchError((dynamic error) {
```

Comment:
This code's hard to parse (`Future.wait(...).catchError(...).then(..., onError: ...)`), which makes it hard to immediately see if we should be waiting for this here.

Can you refactor the inlined code herein into methods to make this a little more readable?

Reply:
I think it's more that the formatting was unhelpful for readability. How is it now?